pallets/parachain-staking/src/lib.rs

```diff
@@ -614,6 +684,37 @@ pub mod pallet {
 			Ok(().into())
 		}
 		#[pallet::weight(0)]
+		/// Set the total number of collator candidates selected per round
+		pub fn set_total_selected(origin: OriginFor<T>, new: u32) -> DispatchResultWithPostInfo {
+			frame_system::ensure_root(origin)?;
+			ensure!(
+				new >= T::MinSelectedCandidates::get(),
+				Error::<T>::CannotSetBelowMin
+			);
+			let old = <TotalSelected<T>>::get();
+			<TotalSelected<T>>::put(new);
+			Self::deposit_event(Event::SetTotalSelected(old, new));
+			Ok(().into())
+		}
+		#[pallet::weight(0)]
+		/// Set blocks per round
+		/// - if called with `new` less than length of current round, will transition immediately
+		/// in the next block
```

Comment:
Maybe an example would be useful?

This round started 20 block ago and is scheduled to last for 40 blocks. Then the xt is called and lowers the blocks_per_round to 30.

This round started 20 blocks ago and is scheduled to last 40 blocks. Then this xt is called and lowers the blocks_per_round to 10.

Actually, those would make good unit tests.

EDIT: I see you already have the second one covered.

Reply:
I covered the first one in a test just added. It will wait 10 more blocks until 30 blocks have passed to jump to the next round.